src/EPPlus/FormulaParsing/Excel/Functions/Finance/Implementations/OddfpriceImpl.cs

```diff
@@ -0,0 +1,225 @@
+
+  A commercial license to use this software can be purchased at https://epplussoftware.com
+ *************************************************************************************************
+  Date               Author                       Change
+ *************************************************************************************************
+  26/06/2023         EPPlus Software AB           EPPlus v7
+ *************************************************************************************************/
+
+using OfficeOpenXml.FormulaParsing.Excel.Functions.Finance.FinancialDayCount;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace OfficeOpenXml.FormulaParsing.Excel.Functions.Finance.Implementations
+{
+    internal class OddfpriceImpl
+    {
+        private readonly System.DateTime _settlementDate;
+        private readonly System.DateTime _maturityDate;
+        private readonly System.DateTime _issueDate;
+        private readonly System.DateTime _firstCouponDate;
+        private readonly double _rate;
+        private readonly double _yield;
+        private readonly double _redemption;
+        private readonly int _frequency;
+        private readonly DayCountBasis _basis;
+
+        public OddfpriceImpl(System.DateTime settlementDate, System.DateTime maturityDate, System.DateTime issueDate, System.DateTime firstCouponDate, double rate, double yield, double redemption, int frequency, DayCountBasis basis)
+        {
+            _settlementDate = settlementDate;
+            _maturityDate = maturityDate;
+            _issueDate = issueDate;
+            _firstCouponDate = firstCouponDate;
+            _rate = rate;
+            _yield = yield;
+            _redemption = redemption;
+            _frequency = frequency;
+            _basis = basis;
+        }
+
+        public FinanceCalcResult<double> GetOddfprice()
+        {
+
+            if (!((_maturityDate > _firstCouponDate)
+               || (_maturityDate > _settlementDate)
+               || (_maturityDate > _issueDate))
+               || !((_firstCouponDate > _settlementDate)
+               || (_firstCouponDate > _issueDate)
+               || (_settlementDate > _issueDate)))
+            {
+                return new FinanceCalcResult<double>(eErrorType.Num);
+            }
+
+            if (_frequency != 1 && _frequency != 2 && _frequency != 4)
+            {
+                return new FinanceCalcResult<double>(eErrorType.Num);
+            }
+            var sDate = FinancialDayFactory.Create(_settlementDate, _basis);
+            var mDate = FinancialDayFactory.Create(_maturityDate, _basis);
+            var fcDate = FinancialDayFactory.Create(_firstCouponDate, _basis);
+            var iDate = FinancialDayFactory.Create(_issueDate, _basis);
+            var daysDefinition = FinancialDaysFactory.Create(_basis);
+
+            var A = daysDefinition.GetDaysBetweenDates(iDate, sDate);
```

Comment:
Some of the variable names below are not that describing, like A, E, N, DFC, T1,T2,T3,T4 etc. Is it possible to give them more reader friendly names? Optionally add a comment describing what they are.

Reply:
Added comment that refers to the formula in the Microsoft documentation

Reply:
Great!